src/protocols/ssh/click.c

```diff
@@ -0,0 +1,457 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+#include "config.h"
+
+#include "common/cursor.h"
+#include "common/display.h"
+#include "common/recording.h"
+#include "click.h"
+#include "ssh.h"
+#include "terminal/terminal.h"
+#include "terminal/select.h"
+
+#include <guacamole/client.h>
+#include <guacamole/unicode.h>
+#include <guacamole/user.h>
+#include <libssh2.h>
+
+#include <pthread.h>
+
+void guac_click_draw_select(guac_click* click){
+
+    int height = click->term->display->char_height;
+    int width = click->term->display->char_width;
+
+    guac_protocol_send_rect(click->socket, click->select_layer, 
+    click->select_head * width, click->select_row * height,
+    (click->select_tail - click->select_head + 1) * width, height);
+
+    guac_protocol_send_cfill(click->socket, GUAC_COMP_SRC, click->select_layer,
+    0x00, 0x80, 0xFF, 0x60);
+
+}
+
+void guac_click_draw_blank(guac_click* click){
+
+    guac_protocol_send_rect(click->socket, click->select_layer, 0, 0, 1, 1);
+
+    guac_protocol_send_cfill(click->socket, GUAC_COMP_SRC, click->select_layer,
+    0x00, 0x00, 0x00, 0x00);
```

Comment:
Why does the double/triple click behavior need its own dedicated selection redraw code, its own dedicated select layer, etc.? Can the existing terminal selection code not be used as-is?